cmd/upgrade/roles/cmd.go

```diff
@@ -0,0 +1,1012 @@
+	isOperatorPolicyUpgradeNeeded, err = r.AWSClient.IsUpgradedNeededForOperatorRolePoliciesUsingCluster(
+		cluster,
+		r.Creator.AccountID,
+		policyVersion,
+		credRequests,
+		operatorRolePolicyPrefix,
+	)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	version := args.upgradeVersion
+
+	missingRolesInCS, err := r.OCMClient.FindMissingOperatorRolesForUpgrade(cluster, version)
+	if err != nil {
+		return err
+	}
+
+	if len(missingRolesInCS) <= 0 && !isOperatorPolicyUpgradeNeeded {
+		if !args.isInvokedFromClusterUpgrade {
+			r.Reporter.Infof(
+				"Operator roles/policies associated with the cluster '%s' are already up-to-date.",
+				cluster.ID(),
+			)
+		}
+		os.Exit(0)
+	}
+
+	if spin != nil {
+		spin.Stop()
+	}
+
+	operatorRolePolicies, err := r.OCMClient.GetPolicies("OperatorRole")
+	if err != nil {
+		r.Reporter.Errorf("Expected a valid role creation mode: %s", err)
+		os.Exit(1)
+	}
+
+	if isOperatorPolicyUpgradeNeeded {
+		err = upgradeOperatorPolicies(
+			args.isInvokedFromClusterUpgrade,
+			mode,
+			r,
+			isUpgradeNeedForAccountRolePolicies,
+			operatorRolePolicies,
+			env,
+			policyVersion,
+			credRequests,
+			cluster,
+			operatorRolePolicyPrefix,
+		)
+		if err != nil {
+			r.Reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+	}
+
+	if len(missingRolesInCS) > 0 {
+		createdMissingRoles := 0
+		for _, operator := range missingRolesInCS {
+			roleName := roles.GetOperatorRoleName(cluster, operator)
+			exists, _, err := r.AWSClient.CheckRoleExists(roleName)
+			if err != nil {
+				return r.Reporter.Errorf("Error when detecting checking missing operator IAM roles %s", err)
+			}
+			if !exists {
+				err = createOperatorRole(
+					mode,
+					r,
+					cluster,
+					missingRolesInCS,
+					args.isInvokedFromClusterUpgrade,
+					operatorRolePolicies,
+					operatorRolePath,
+					operatorPolicyPath,
+					operatorRolePolicyPrefix,
+				)
+				if err != nil {
+					r.Reporter.Errorf("%s", err)
+					os.Exit(1)
+				}
+				createdMissingRoles++
+			}
+		}
+		if createdMissingRoles == 0 {
+			r.Reporter.Infof(
+				"Missing roles/policies have already been created. Please continue with cluster upgrade process.",
+			)
+		}
+	}
+	return err
+}
+
+func LogError(key string, ocmClient *ocm.Client, defaultPolicyVersion string, err error, reporter *rprtr.Object) {
+	reporter.Debugf("Logging throttle error")
+	if strings.Contains(err.Error(), "Throttling") {
+		ocmClient.LogEvent(key, map[string]string{
+			ocm.Response:   ocm.Failure,
+			ocm.Version:    defaultPolicyVersion,
+			ocm.IsThrottle: "true",
+		})
+	}
+}
+
+func handleAccRolePolicyARN(
+	mode string,
+	awsClient aws.Client,
+	roleName string,
+	rolePath string,
+	accountID string,
+) (string, []string, error) {
+	commands := make([]string, 0)
+	policiesDetails, err := awsClient.GetAttachedPolicy(&roleName)
+	if err != nil {
+		return "", commands, err
+	}
+
+	hasMoreThanOneAttachedPolicy := aws.CountAttachedPoliciesInDetails(policiesDetails) > 1
```

Comment:
If I'm not wrong we are only using this function to check if `hasMoreThanOneAttachedPolicy`.
Should we change the function to return `true` in case there is an attached policy instead of counting?